app_feup/lib/controller/networking/network_router.dart

```diff
@@ -177,31 +178,56 @@ class NetworkRouter {
 
   static Future<List<Trip>> getNextArrivalsStop(
       String stopCode, BusStopData stopData) async {
-    final String url =
-        'http://move-me.mobi/NextArrivals/GetScheds?providerName=STCP&stopCode=STCP_' +
-            stopCode;
+
+    final url =
+        'https://www.stcp.pt/pt/itinerarium/soapclient.php?codigo=' + stopCode;
+
     final http.Response response = await http.get(url);
-    final List<Trip> tripList = List();
+    final htmlResponse = parse(response.body);
 
-    final List json = jsonDecode(response.body);
+    final tableEntries =
+        htmlResponse.querySelectorAll('#smsBusResults > tbody > tr.even');
+
+    final configuredBuses = stopData.configuredBuses;
+
+    final tripList = List<Trip>();
+
+    for (var entry in tableEntries) {
+      final info = entry.querySelectorAll('td');
 
-    for (var TripKey in json) {
-      final trip = TripKey['Value'];
-      final String line = trip[0];
-      if (stopData.configuredBuses.contains(line)) {
-        final String destination = trip[1];
-        String timeString = trip[2];
-        if (timeString.substring(timeString.length - 1) == '*') {
-          timeString = timeString.substring(0, timeString.length - 1);
+      final line = info[0].querySelector('ul > li').text.trim();
+
+      if(!configuredBuses.contains(line)){
+        continue;
+      }
+
+      var destination = info[0].text
+          .replaceAll('\n', '')
+          .replaceAll('\t', '')
+          .replaceAll(' ', '')
+          .replaceAll('-', '');
+
+      destination = destination.substring(line.length + 1);
+
+      final timeOfArrival = info[1].text.trim();
+      var timeRemaining = '0';
+
+      if(timeOfArrival != 'a passar'){
+        timeRemaining = info[2].text;
+        if(timeRemaining.contains('min')){
+          timeRemaining = timeRemaining.substring(0, timeRemaining.length - 3);
         }
-        final int timeRemaining = int.parse(timeString);
-        final Trip newTrip = Trip(
-            line: line, destination: destination, timeRemaining: timeRemaining);
-        tripList.add(newTrip);
       }
```

Comment:
Maybe we can extract this logic into another function to make it more readable?

Reply:
When extracting the logic we can already guarantee that it is an integer as well, I think

Reply:
Sure, I will take care of it next commit.